.github/actions/openvino_provider/action.yml

```diff
@@ -0,0 +1,156 @@
+name: 'OpenVINO provider'
+description: 'Provides OpenVINO build artifacts from a requested source'
+inputs:
+  revision:
+    description: 'Version of OpenVINO to provide - latest_available | HEAD | specific commit hash'
+    required: true
+  branch_name:
+    description: 'Branch of OpenVINO to take the revision from if no specific hash was provided. 
+                  Taken from github context by default'
+    required: false
+  ov_artifact_name:
+    description: "Name under which to upload provided OpenVINO build artifacts, set automatically by default"
+    required: false
+  commit_packages_to_provide:
+    description: "Comma-separated list of OpenVINO packages from post-commit to provide, all available by default"
+    required: false
+  commit_build_key:
+    description: 'public_linux_ubuntu_20_04_release | public_linux_ubuntu_22_04_release | 
+                  public_linux_ubuntu_24_04_release | public_windows_vs2019_release'
+    required: false
+  commit_share_path:
+    description: 'Path to share holding OpenVINO post-commit artifacts'
+    required: false
+    default: '/mount/build-artifacts'
+  nightly_build_key:
+    description: 'centos7_x86_64 | debian9_armhf | rhel8_x86_64 | ubuntu20_arm64 | ubuntu20_x86_64 | ubuntu22_x86_64 | 
+                  ubuntu24_x86_64 | macos_12_6_arm64 | macos_12_6_x86_64 | windows_x86_64'
+    required: false
+  nightly_package_source:
+    description: 'Source URL hosting OpenVINO nightly archives'
+    required: false
+    default: 'https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/latest.json'
+  nightly_pip_extra_url:
+    description: 'Source URL hosting OpenVINO nightly wheels, for passing to pip as --extra-index-url'
+    required: false
+    default: 'https://storage.openvinotoolkit.org/simple/wheels/nightly'
+outputs:
+  ov_artifact_name:
```

Comment:
can restore artifacts also download it ? now each restore action is followed by download step

Reply:
In usage example I executed both openvino-provider action and artifact download within the same job just for simplicity; but in the real-world scenario they could be in separate jobs (e.g. openvino-provider is executed on Azure-based runner with an access to the Azure shared drive to fetch the artifacts and upload them to GitHub; while the actual build job is executed on GitHub-based runner and downloads the artifacts from GitHub - this way we don't need to migrate build jobs to Azure-based runners in tokenizers, for example.

If both provider and build job are executed on the same runner, download step is not required, since the artifacts are already in the runner's workspace (I'll test this scenario separately)